crates/nu-parser/src/flatten.rs

```diff
@@ -92,44 +92,91 @@ impl Display for FlatShape {
     }
 }
 
-pub fn flatten_block(working_set: &StateWorkingSet, block: &Block) -> Vec<(Span, FlatShape)> {
-    let mut output = vec![];
-
+fn flatten_block_inner(
```

Comment:
Maybe worth adding a doccomment explaining the rationale right in the code for the `_inner` fn's

Reply:
Thanks! (I personally inclined to put stuff like this even in the doccomments so you see it if you try to use it and it is local to each function, but I am fine with your block comment as well)